postgres-protocol/Cargo.toml

```diff
@@ -19,3 +19,6 @@ memchr = "2.0"
 rand = "0.8"
 sha2 = "0.10"
 stringprep = "0.1"
+
+[features]
+sync-only = []
```

Comment:
This is still a back compat break. Any crate that enables this will affect all other consumers of postgres in the crate graph.

Reply:
This is why it's an optional feature. It is intended to be not fully compatible with other versions and has to explicitly defined. 

I cannot see other way, except creating a structures variants and then selecting them via call to the proper method (like `execute`, `execute_sync`). 

The use case for it is also very specific - you need to pass connection with a sync requirement

Reply:
Features are global across all consumers of the library in a crate graph.

Reply:
I understand your concerns. I will adjust to your will :)